src/lib.rs

```diff
@@ -99,3 +104,117 @@ impl fmt::Display for EmailAddress {
         }
     }
 }
+
+#[derive(Debug, PartialEq)]
+pub enum EmailAddressError {
+    InvalidDisplayName,
+    InvalidEmailAddress,
+}
+
+impl fmt::Display for EmailAddressError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            EmailAddressError::InvalidDisplayName => write!(f, "Invalid display name"),
+            EmailAddressError::InvalidEmailAddress => write!(f, "Invalid email address"),
+        }
+    }
+}
+
+impl std::error::Error for EmailAddressError {}
+
+/// Basic validation of display name.
+fn is_valid_display_name(name: &str) -> bool {
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"^[^<>]+$").unwrap();
+    }
+    RE.is_match(name)
+}
+
+/// Basic validation of address.
+fn is_valid_address(address: &str) -> bool {
+    lazy_static! {
+        // TODO: use a proper regex
+        static ref RE: Regex = Regex::new(r"^[^<> ]+@[^<> ]+\.[^<> ]+$").unwrap();
+    }
+    RE.is_match(address)
+}
+
+impl<'a> TryFrom<&'a str> for EmailAddress {
```

Comment:
Can also do

```rust
impl<T> TryFrom<T> for EmailAddress where T: AsRef<str>{ 
    ...
}
```

So that when dealing with a `String` or something there's no need to explicitly take the reference

Reply:
Yeah right, can change now.